docs: Dropdown migration to v4 (#1508)
contentful/forma-36: MIGRATION.md

  'top-right': 'top-end',
}
```

### How to migrate your Dropdown components

The migration should be done manually since there are different approach for each specific case.
- Replace it with `Menu` if it was used to offer a list of choices to the user, such as a set of actions or links. 
- Replace it with `Autocomplete`  if it was used in combination with input to help user set the value by chosing from the list of options.
- Replace it with `Popover` if it was used for some custom case that does not match the cases described above. But mostly you don't need to use `Popover`. *Note: `Popover` is a low level component and it is used as a base for `Autocomplete` and `Menu`*

Below you will find a couple of examples how to migrate `Dropdown` component:

#### `Dropdown` to `Menu`

**Simple case:**

Importent things to take into account in the following example:
- By default `Menu` is uncontrolled component, so you don't have to pass callbacks and state. Nevertheless you can make it controlled if you need. [See the controlled Menu example in component docs](https://github.com/contentful/forma-36/blob/forma-v4/packages/components/menu/Menu.mdx#controlled-menu)

> - By default click on `MenuItem` will close menu, if you want to disable this behaviour, just pass prop `closeOnSelect="false"`

Reviewer: are we doing American English?

> - By default click on `MenuItem` will close menu, if you want to disable this behaviour, just pass prop `closeOnSelect="false"`

Author: I don't think we have any preferences, but I don't mind to change it 🙂

        setDropdownOpen((isOpen) => !isOpen)
      }}
    />
  }>
  <DropdownList maxHeight={200}>
    <DropdownListItem isTitle>Actions</DropdownListItem>
    <DropdownListItem>
      First action
    </DropdownListItem>
    <DropdownListItem>
      Second action
    </DropdownListItem>
    <DropdownList border="top">
      <DropdownListItem>
        <TextLink href="https://contentful.com" target="_blank">About Contentful</TextLink>
      </DropdownListItem>
    </DropdownList>
  </DropdownList>
</Dropdown>
```

Forma v4:

```tsx static=true
import { Menu, IconButton } from '@contentful/f36-components';

<Menu>
  <Menu.Trigger>
    <IconButton
      aria-label="Open Actions"
      icon={<MoreVerticalIcon size="small" />}
    />
  </Menu.Trigger>
  <Menu.List style={{ maxHeight: '200px' }}>
    <Menu.SectionTitle>Actions</Menu.SectionTitle>
    <Menu.Item>First action</Menu.Item>
    <Menu.Item>Second action</Menu.Item>
    <Menu.Divider />
    <Menu.Item as="a" href="https://contentful.com" target="_blank">
      About Contentful
    </Menu.Item>
  </Menu.List>
</Menu>
```

**Case with Submenu:**
Importent things to take into account in the following example:

> - You can't just pass `Dropdown` as an Menu item. 

Reviewer: I think it's because of how it was done previously, not you can't do it.

> - You can't just pass `Dropdown` as an Menu item. 

Author: I've rephrased it a bit. Please take a look.

> - You have to use specific components like `Menu.Submenu` and `Menu.SubmenuTrigger` to add submenu, but other components like `Menu.List`, `Menu.Item`, `Menu.Divider`, etc. remain the same.

Author: Yeah, I've rephrased, please take a look.

          Submenu action 1
        </DropdownListItem>
        <DropdownListItem>
          Submenu action 2
        </DropdownListItem>
      </DropdownList>
    </Dropdown>
    <DropdownListItem>
      Second action
    </DropdownListItem>
  </DropdownList>
</Dropdown>
```

Forma v4:

```tsx static=true
import { Menu, IconButton } from '@contentful/f36-components';

<Menu>
  <Menu.Trigger>
    <IconButton
      aria-label="Open Actions"
      icon={<MoreVerticalIcon size="small" />}
    />
  </Menu.Trigger>
  <Menu.List>
    <Menu.Item onClick={() => {/* do some action */}}>First action</Menu.Item>
    <Menu.Submenu>
      <Menu.SubmenuTrigger>Open submenu</Menu.SubmenuTrigger>
      <Menu.List>
        <Menu.Item>Submenu action 1</Menu.Item>
        <Menu.Item>Submenu action 2</Menu.Item>
      </Menu.List>
    </Menu.Submenu>
    <Menu.Item onClick={() => {/* do some action */}}>Second action</Menu.Item>
  </Menu.List>
</Menu>
```

#### `Dropdown` to `Popover`

**Popover case with checkboxes:**

Importent things to take into account in the following example:
- Unlike `Menu`, `Popover` is a controlled component. So you have to pass callbacks and state in order to make it work.

> - We used `FocusLock` from [react-focus-lock](https://github.com/theKashey/react-focus-lock) to enhance a11y. So user can't leave the popover with "Tab" navigation.

Reviewer 2: ```suggestion
- We used `FocusLock` from [react-focus-lock](https://github.com/theKashey/react-focus-lock) to enhance a11y. It traps users' focus in the popover context, user will not be able to leave the component when using "Tab" navigation. 
```

Reviewer 2: I wanted to explain a bit more how it works, but I don't know if I did a good job here 😬

  <Button variant="primary" size="small" />
</Modal.Controls>;
```

## Dropdown

> The Dropdown component is replaced with a couple of new components: [Menu](https://github.com/contentful/forma-36/blob/forma-v4/packages/components/menu/Menu.mdx), [Autocomplete](https://github.com/contentful/forma-36/blob/forma-v4/packages/components/autocomplete/Autocomplete.mdx) and [Popover](https://github.com/contentful/forma-36/blob/forma-v4/packages/components/popover/Popover.mdx).

Reviewer 3: We could just say **three new components** now that I think about it.

      First action
    </DropdownListItem>
    <DropdownListItem>
      Second action
    </DropdownListItem>
    <DropdownList border="top">
      <DropdownListItem>
        <TextLink href="https://contentful.com" target="_blank">About Contentful</TextLink>
      </DropdownListItem>
    </DropdownList>
  </DropdownList>
</Dropdown>
```

Forma v4:

```tsx static=true
import { Menu, IconButton } from '@contentful/f36-components';

<Menu>
  <Menu.Trigger>
    <IconButton
      aria-label="Open Actions"
      icon={<MoreVerticalIcon size="small" />}
    />
  </Menu.Trigger>
  <Menu.List style={{ maxHeight: '200px' }}>
    <Menu.SectionTitle>Actions</Menu.SectionTitle>
    <Menu.Item>First action</Menu.Item>
    <Menu.Item>Second action</Menu.Item>
    <Menu.Divider />
    <Menu.Item as="a" href="https://contentful.com" target="_blank">
      About Contentful
    </Menu.Item>
  </Menu.List>
</Menu>;
```

**Case with Submenu:**

Keep in mind:

- Instead of nesting another `Dropdown` you should use specific component `Menu.Submenu`.
- Instead of passing `submenuToggleLabel` prop you should use specific component `Menu.SubmenuTrigger`. That will serve as a trigger for a submenu.

> - Submenu will be always shown on the right side of the menu. There is no way to show it on the left side right now.

Reviewer 3: ```suggestion
- The submenu is always displayed on the right side of the menu. There is no way to display it on the left side now.
```

      <DropdownList>
        <DropdownListItem>
          Submenu action 1
        </DropdownListItem>
        <DropdownListItem>
          Submenu action 2
        </DropdownListItem>
      </DropdownList>
    </Dropdown>
    <DropdownListItem>
      Second action
    </DropdownListItem>
  </DropdownList>
</Dropdown>
```

Forma v4:

```tsx static=true
import { Menu, IconButton } from '@contentful/f36-components';

<Menu>
  <Menu.Trigger>
    <IconButton
      aria-label="Open Actions"
      icon={<MoreVerticalIcon size="small" />}
    />
  </Menu.Trigger>
  <Menu.List>
    <Menu.Item
      onClick={() => {
        /* do some action */
      }}
    >
      First action
    </Menu.Item>
    <Menu.Submenu>
      <Menu.SubmenuTrigger>Open submenu</Menu.SubmenuTrigger>
      <Menu.List>
        <Menu.Item>Submenu action 1</Menu.Item>
        <Menu.Item>Submenu action 2</Menu.Item>
      </Menu.List>
    </Menu.Submenu>
    <Menu.Item
      onClick={() => {
        /* do some action */
      }}
    >
      Second action
    </Menu.Item>
  </Menu.List>
</Menu>;
```

#### `Dropdown` to `Popover`

**Popover case with checkboxes:**

Keep in mind:

> - Unlike `Menu`, `Popover` is a controlled component. So you have to pass callbacks and state in order to make it work like expected.

Reviewer 3: ```suggestion
- Unlike `Menu`, `Popover` is a controlled component. So you have to pass callbacks and state to make it work as expected.
```

> - We used `FocusLock` from [react-focus-lock](https://github.com/theKashey/react-focus-lock) to enhance a11y. It traps users' focus in the popover content, user will not be able to leave the popover when using "Tab" navigation.

Reviewer 3: ```suggestion
- We used `FocusLock` from [react-focus-lock](https://github.com/theKashey/react-focus-lock) to enhance a11y. It traps users' focus in the popover content and they will not be able to leave the popover when using "Tab" navigation.
```
